ui/components/app/transaction-list/transaction-list.component.js

```diff
@@ -123,23 +123,36 @@ export default function TransactionList({
             <div className="transaction-list__header">
               {`${t('queue')} (${pendingTransactions.length})`}
             </div>
-            {pendingTransactions.map((transactionGroup, index) =>
-              transactionGroup.initialTransaction.transactionType ===
-              TransactionType.smart ? (
-                <SmartTransactionListItem
-                  isEarliestNonce={index === 0}
-                  smartTransaction={transactionGroup.initialTransaction}
-                  transactionGroup={transactionGroup}
-                  key={`${transactionGroup.nonce}:${index}`}
-                />
-              ) : (
-                <TransactionListItem
-                  isEarliestNonce={index === 0}
-                  transactionGroup={transactionGroup}
-                  key={`${transactionGroup.nonce}:${index}`}
-                />
-              ),
-            )}
+            {pendingTransactions
+              ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+              .sort(
+                (a, b) => b.primaryTransaction.time - a.primaryTransaction.time,
+              )
+              ///: END:ONLY_INCLUDE_IN
+              .map((transactionGroup, index) => {
+                ///: BEGIN:ONLY_INCLUDE_IN(build-main,build-beta,build-flask)
+                if (
+                  transactionGroup.initialTransaction.transactionType ===
+                  TransactionType.smart
+                ) {
+                  return (
+                    <SmartTransactionListItem
+                      isEarliestNonce={index === 0}
+                      smartTransaction={transactionGroup.initialTransaction}
+                      transactionGroup={transactionGroup}
+                      key={`${transactionGroup.nonce}:${index}`}
+                    />
+                  );
+                }
+                ///: END:ONLY_INCLUDE_IN
```

Comment:
Is this exclusion because MMI doesn't support smart transactions?

Reply:
Yes! But I also think that we might not need that code fence, because the user will not be able even start a Smart Transaction... so that IF will always be false I would say.